src/webgpu/shader/validation/v_0033.spec.ts

```diff
@@ -0,0 +1,72 @@
+export const description = `
+Tests for validation rule v-0033:
+If present, the initializer’s type must match the store type of the variable.
+`;
+
+import { params, poptions } from '../../../common/framework/params_builder.js';
+import { makeTestGroup } from '../../../common/framework/test_group.js';
+import { assert } from '../../../common/framework/util/util.js';
+
+import { ShaderValidationTest } from './shader_validation_test.js';
+
+export const g = makeTestGroup(ShaderValidationTest);
+
+type ScalarType = 'f32' | 'i32' | 'u32' | 'bool';
+const kScalarType = ['i32', 'f32', 'u32', 'bool'] as const;
+
+function getTypeInfo(scalarType: ScalarType, x: number, y: number) {
```

Comment:
uNit: Since you've simplified this to not have to return the zero values (nice! I totally didn't think of that...), a better name could just be `getType`, `getTypeString`, `getTypeForMatrixVectorScalar`, or something like that.

OTOH, this function could go away; see below.